File: MobiFon.Core/Entities/Person.cs
using MobiFon.Core.Entities.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;
using MobiFon.Core.Enumerations;

namespace MobiFon.Core.Entities
{
    public class Person: BaseEntity
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime BirthDate { get; set; }
        public Gender? Gender { get; set; }
        public string? ProfilePhoto { get; set; }
        public string? ProfilePhotoThumbnail { get; set; }
        public int? BirthPlaceId { get; set; }
        public City? BirthPlace { get; set; }
        public string? JMBG { get; set; }
        public string? Qualifications { get; set; }
        public int? PlaceOfResidenceId { get; set; }
        public City? PlaceOfResidence { get; set; }
        public MarriageStatus? MarriageStatus { get; set; }
        public string? Nationality { get; set; }
        public string? Citizenship { get; set; }
        public bool WorkExperience { get; set; }
        public string Address { get; set; }
        public string PostCode { get; set; }
        public string? Biography { get; set; }
        public Position? Position { get; set; }
        public DateTime? DateOfEmployment { get; set; }
        public float? Pay { get; set; }
        public bool MembershipCard { get; set; }
        public int ApplicationUserId { get; set; }
    }
}

File: UI/propertease_admin/lib/models/person.dart
import 'package:json_annotation/json_annotation.dart';

import 'city.dart';

part 'person.g.dart';

@JsonSerializable()
class Person {
  int? id;
  DateTime? createdAt;
  DateTime? modifiedAt;
  int? totalRecordsCount;
  bool? isDeleted;
  String? firstName;
  String? lastName;
  DateTime? birthDate;
  int? gender; // Enum values are stored as integers
  String? genderName;
  String? profilePhoto;
  String? profilePhotoThumbnail;
  int? birthPlaceId;
  City? birthPlace;
  String? jmbg;
  String? qualifications;
  int? placeOfResidenceId;
  City? placeOfResidence;
  int? marriageStatus; // Enum values are stored as integers
  String? marriageStatusName;
  String? nationality;
  String? citizenship;
  bool? workExperience;
  String? address;
  String? postCode;
  String? biography;
  int? position;
  String? positionName;
  DateTime? dateOfEmployment;
  double? pay;
  bool? membershipCard;
  int? applicationUserId;

  Person();

  factory Person.fromJson(Map<String, dynamic> json) => _$PersonFromJson(json);
  Map<String, dynamic> toJson() => _$PersonToJson(this);
}

enum Gender { male, female, other }

enum MarriageStatus { single, married, divorced, widowed }
